www/components/shelf/shelf.js

```diff
@@ -99,6 +83,32 @@ class Shelf extends LitElement {
     this.requestUpdate();
   }
 
+  async fetchShelfData() {
+    console.log('ENTER fetchShelfData');
+    const response = await client.query({
+      query: MenuQuery,
+      variables: {
+        name: 'side',
+        route: `/${this.page}/`,
+        order: 'index_asc'
+      }
+    });
+
+    this.shelfList = response.data.menu.children;
+    console.log('fetchShelfData - ', this.shelfList);
+  }
+
+  async updated(changedProperties) {
+    console.log('ENTER updated - changedProperties', changedProperties);
+    console.log('updated - this.page', this.page);
+    if (changedProperties.has('page') && this.page !== '' && this.page !== '/') {
+      await this.fetchShelfData();
```

Comment:
You're expecting to return a promise that doesn't exist.  Maybe just return the query(which is a promise) like this:

```js
const response = await this.fetchShelfData();
this.shelfList = response.data.menu.children;
```

Reply:
still happening, I guess I should just revert back the `connectedCallback` / `window.location.pathname` approach then? 

Not sure if we'll need to start recommending / documenting certain patterns around this stuff then, though I would hope we don't have to be any different than SPA development.  Like if having unpredictable lifecycles could be a concern, or just needing to guard initial values of your variables.